go/pkg/project/project.go

```diff
@@ -86,29 +169,157 @@ func (p *Project) CheckpointOrExperimentFromPrefix(prefix string) (*CheckpointOr
 	return matches[0], nil
 }
 
-func (p *Project) DeleteCheckpoint(com *Checkpoint) error {
-	if err := p.repository.Delete(com.StorageTarPath()); err != nil {
-		console.Warn("Failed to delete checkpoint storage directory %s: %s", com.StorageTarPath(), err)
+func (p *Project) DeleteCheckpoint(chk *Checkpoint) error {
+	if err := p.repository.Delete(chk.StorageTarPath()); err != nil {
+		console.Warn("Failed to delete checkpoint storage directory %s: %s", chk.StorageTarPath(), err)
 	}
+	p.invalidateCache()
 	return nil
 }
 
 func (p *Project) DeleteExperiment(exp *Experiment) error {
+	console.Debug("Deleting experiment: %s", exp.ShortID())
 	if err := p.repository.Delete(exp.HeartbeatPath()); err != nil {
 		console.Warn("Failed to delete heartbeat file %s: %s", exp.HeartbeatPath(), err)
 	}
 	if err := p.repository.Delete(exp.StorageTarPath()); err != nil {
-		console.Warn("Failed to delete checkpoint storage directory %s: %s", exp.StorageTarPath(), err)
+		console.Warn("Failed to delete experiment storage directory %s: %s", exp.StorageTarPath(), err)
 	}
 	if err := p.repository.Delete(exp.MetadataPath()); err != nil {
 		console.Warn("Failed to delete experiment metadata file %s: %s", exp.MetadataPath(), err)
 	}
+	p.invalidateCache()
 	return nil
 }
 
+type CreateExperimentArgs struct {
+	Path           string
+	Command        string
+	Params         map[string]param.Value
+	PythonPackages map[string]string
+}
+
+func (p *Project) CreateExperiment(args CreateExperimentArgs, async bool, workChan chan func() error) (*Experiment, error) {
+	spec, err := repository.LoadSpec(p.repository)
+	if err != nil {
+		return nil, err
+	}
+	if spec == nil {
+		if err := repository.WriteSpec(p.repository); err != nil {
+			return nil, err
+		}
+	} else if spec.Version > repository.Version {
+		return nil, errors.IncompatibleRepositoryVersion(p.repository.RootURL())
+	}
+
+	hostIP, err := localIP()
```

Comment:
This is changing behavior I think isn't it? I think this got bumped because we wanted to come up with some sensible behavior for localhost, IIRC. #203

Reply:
I think this function always will return local addresses, so maybe the solution is to just blank it out for now. It's a shame since it's useful information when you're running on multiple hosts, but I can't think of a good solution.